app/src/main/java/com/csai/swanton/tasks/ConnectTask.java

```diff
@@ -0,0 +1,53 @@
+package com.csai.swanton.tasks;
+
+import android.os.AsyncTask;
+import android.util.Log;
+import android.view.View;
+import com.csai.swanton.sftp.ChannelSftpFactory;
+import com.csai.swanton.util.ActivityTags.Tags;
+import com.google.android.material.snackbar.Snackbar;
+import com.jcraft.jsch.ChannelSftp;
+import com.jcraft.jsch.JSchException;
+import java.lang.ref.WeakReference;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import lombok.AllArgsConstructor;
+
+/**
+ * Asynchronous task for connecting to a raspberry pi.
+ */
+@AllArgsConstructor
+public class ConnectTask extends AsyncTask<Void, Void, Optional<ChannelSftp>> {
+
+  @Nonnull
+  private final WeakReference<View> view;
+
+  @Override
+  protected void onPreExecute() {
+    Snackbar.make(this.view.get(), "Connecting to Raspberry Pi...", Snackbar.LENGTH_SHORT).show();
+    Log.i(Tags.MAIN_ACTIVITY.getTag(), "Connecting to Raspberry Pi...");
+  }
+
+  @Override
+  protected Optional<ChannelSftp> doInBackground(final Void... params) {
+    try {
+      return Optional.of(ChannelSftpFactory.create());
+    } catch (final JSchException e) {
+      // Swallow exception here and log in onPostExecute so app doesn't crash
+      return Optional.empty();
+    }
+  }
+
+  @Override
+  protected void onPostExecute(final Optional<ChannelSftp> channelSftp) {
+    if (channelSftp.isPresent()) {
+      Snackbar.make(
+          this.view.get(), "Connected to Raspberry Pi!", Snackbar.LENGTH_SHORT).show();
+      Log.i(Tags.MAIN_ACTIVITY.getTag(), "Connected to Raspberry Pi.");
+    } else {
+      Snackbar.make(
+          this.view.get(), "Failed to connect to Raspberry Pi.", Snackbar.LENGTH_SHORT).show();
+      Log.e(Tags.MAIN_ACTIVITY.getTag(), "Failed to connect to Raspberry Pi.");
```

Comment:
Does this include a reason we can't connect? It might be useful to separate out incorrect credentials/can't connect/other errors, or at least include a log of this. Something to think about for later I think, not super critical right now.

Reply:
right now no, but that can and should be changed, I agree!

Reply:
Change was made